addons/hr/models/hr_employee_public.py

```diff
@@ -216,4 +216,10 @@ def init(self):
         )""" % (self._table, self._get_fields()))
 
     def get_avatar_card_data(self, fields):
-        return self.read(fields)
+        values = self.read(fields)
+        for val, record in zip(values, self):
+            if record.user_id:
+                val['show_im_status'] = record.active and record.user_id.active
+            else:
+                val['show_im_status'] = record.active
```

Comment:
Rather than repeating these LOC on server side couldn't we implement this on client side?